### DISCUSSION TITLE
Microsoft Store - WACK validation warnings

### DISCUSSION THREAD

**Discussion by MalacTheLittle** [airsdk/Adobe-Runtime-Support#2436] [Q&A]
I get warnings from the Windows App Certification Kit (when checking .msix packages) related to DPI-awareness, NT services and running executables.

@ajwfrost Can you please check attached report and tell if there's a way to fix issues, especially the last one (launching executables) since it disables app to install on Windows 10 S systems.

Thanks
[Windows App Certification Kit - Test Results_xml.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/10456336/Windows.App.Certification.Kit.-.Test.Results_xml.zip)

**Comment by ajwfrost** [comment]
Hi

Belatedly, sorry:

> nor calls into DPI Awareness APIs for ex: user32!SetProcessDpiAwarenessContext or user32!SetThreadDpiAwarenessContext.

Well actually, it does, but not in a way that they can detect because it has to be flexible enough to cope with different Windows versions and those calls aren't available in older versions that we support.


> Registry path REGISTRY\MACHINE\SYSTEM\CurrentControlSet\Services\bam|VSS found.

I can't find "bam" or "VSS" but there are a number of registry accesses used for fine-tuning and detection of things especially around media. Not sure that we're writing any NT Services though! We can perhaps try to work out what's actually calling this...?

> File xxx contains a blocked executable reference to  "yyy"

Not sure what the .dat file are that it mentions - Registry.dat, User.dat - presumably these are within your control?
The other ones (eScript.swf and Adobe AIR.dll) may just be false positives e.g. the SWF file having an executable reference...?!

> File Adobe AIR\Versions\1.0\Adobe AIR.dll contains a reference to a "Launch Process" related API kernel32.dll!zzz

These are used for things like opening a file using the default application, and NativeProcess functionality.

It sounds like to fix all of these issues, we'd need to create a special runtime for it .. see also the request in #2016 - this would need a significant amount of work to set up...

**Reply by MalacTheLittle** [reply]
Thanks Andrew! 
In the meantime I have successfully published app to the Store, only thing I had to do is to state that app is not compatible with Windows 10 S system at the top of app description.

> Not sure what the .dat file are that it mentions - Registry.dat, User.dat - presumably these are within your control?

These are created by `MSIX Packaging Tool`, not sure what it's for, the header of the files starts with "regf". Not sure how much are you familiar with this tool, but the process of making .msix installer goes that it records live installation of .msi or .exe installer.

> The other ones (eScript.swf and Adobe AIR.dll) may just be false positives e.g. the SWF file having an executable reference...?!

This was my tought as well, but I'm not referencing any executables within the app, as well as `NativeProcess` or `openWithDefaultApplication`.


If you're interested in getting to the bottom of the problem I can send you .msix file that I've uploaded to Store. 
I have tested the app on Windows 10 S that (somehow) I installed on virtual machine and it worked just fine, but since I'm not familiar with Windows 10 S purpose and functionalities and installation is not straight forward like other Windows installations, I might be wrong.
But if I'm right and it does work than it's a pity that we need to state it's not supported just because of some automated Store tests, instead of having one more platform supported by AIR.
